lib/dry/configurable/setting.rb

```diff
@@ -19,106 +19,26 @@ class Setting
       # @api private
       attr_reader :name
 
-      # @api private
-      attr_reader :writer_name
-
-      # @api private
-      attr_reader :input
-
       # @api private
       attr_reader :default
 
       # @api private
       attr_reader :options
 
-      # Specialized Setting which includes nested settings
-      #
       # @api private
-      class Nested < Setting
-        DEFAULT_CONSTRUCTOR = Config.method(:new)
-
-        # @api private
-        def pristine
-          with(input: input.pristine)
-        end
-
-        # @api private
-        def constructor
-          config_class.equal?(Config) ? DEFAULT_CONSTRUCTOR : config_class.method(:new)
-        end
-
-        # @api private
-        def config_class
-          options[:config_class] || Config
-        end
-      end
+      attr_reader :children
 
       # @api private
       def self.cloneable_value?(value)
         CLONEABLE_VALUE_TYPES.any? { |type| value.is_a?(type) }
       end
 
       # @api private
-      def initialize(name, input: Undefined, default: Undefined, **options)
+      def initialize(name, default: Undefined, children: nil, **options)
```

Comment:
I'd do `children: EMPTY_ARRAY` and then down below instead of `!!children` it'd be `children.empty?`